lib/ansible/runner.rb

```diff
@@ -8,7 +8,12 @@ def run(env_vars, extra_vars, playbook_path)
       private
 
       def run_via_cli(env_vars, extra_vars, playbook_path)
-        result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
+        result = AwesomeSpawn.run(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
+        if result.failure?
+          $log.error("ansible-playbook errored - stdout: #{result.output}")
+          $log.error("ansible-playbook errored - stderr: #{result.error}")
+          raise AwesomeSpawn::CommandResultError
+        end
         JSON.parse(result.output)
```

Comment:
you should be able to parse the failures from here?

btw. we plan to parse it, then put it in a presentable form and send as a notification to user (if it's e.g. a bad value send in a form)

also, we will soon switch to https://github.com/ManageIQ/manageiq/pull/17688

----

I think we should log only error that is not user based (like a bad value in form), @agrare thoughs?

Reply:
@miha-plesko ok, so the runner is a bit different. I will add modification so the .run returns a response object. Where you can get the .return_code, .stdout, .stdout_parsed, .stderr and later some nice error messages, if we can get those in a generic way.

Reply:
Sounds like a plan, thanks 👍